crates/ruff_db/src/diagnostic/mod.rs

```diff
@@ -601,21 +613,95 @@ impl std::fmt::Display for DiagnosticId {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum UnifiedFile {
+    Ty(File),
+    Ruff(SourceFile),
+}
```

Comment:
Can you add docs to this type? This might be a nice place to put some prose about why this exists and the high level strategy for how we're trying to share the diagnostics system between both ty and ruff.